source/fe/fe_system.cc

```diff
@@ -1409,7 +1409,9 @@ FESystem<dim, spacedim>::build_interface_constraints()
 {
   // TODO: the implementation makes the assumption that all faces have the
   // same number of dofs
-  AssertDimension(this->n_unique_faces(), 1);
+  if (this->n_unique_faces() > 1)
+    return;
```

Comment:
Not ideal but we currently don't support hanging-nodes for meshes with wedges/pyramids anyway.

Reply:
Hm, yes, not ideal. How can we catch this error in a better way? Is there a better place where this assertion could be put?